### PR TITLE
Events for txn status changes

### PR DESCRIPTION
**Pull request by brianosaurus**
Emits all transaction status changes.
Debugged failure case for transactions (it works just fine).

Use it like this:

```ts
import { TXEventName } from "@augurproject/sdk/build/constants";
import { TXStatusCallback } from "@augurproject/sdk/events";
import { TXStatus } from "./event-handlers";

const failure: TXStatusCallback = (status: TXStatus) => {};
const pending: TXStatusCallback = (status: TXStatus) => {};
const awaitingSigning: TXStatusCallback = (status: TXStatus) => {};

augur.on(TXEventName.Failure, failure);
augur.on(TXEventName.Pending, pending);
augur.on(TXEventName.AwaitingSigning, awaitingSigning);
```

### CODE REVIEW

File: packages/augur-test/src/tests/api/TransactionStatus.test.ts [hunk 1]
```diff
@@ -1,44 +1,140 @@
-import { ContractAPI, loadSeed, ACCOUNTS } from "@augurproject/tools";
 import { BigNumber } from "bignumber.js";
+import { ContractAPI, loadSeed, ACCOUNTS } from "@augurproject/tools";
+import { SECONDS_IN_A_DAY } from '@augurproject/sdk/build/state/getter/Markets';
```
Comment (robert-e-davidson3) [inline packages/augur-test/src/tests/api/TransactionStatus.test.ts]: We should export this from augur-sdk's index.ts.

File: packages/augur-test/src/tests/api/TransactionStatus.test.ts [hunk 2]
```diff
@@ -1,44 +1,140 @@
...
+
+let john: ContractAPI;
+
+beforeAll(async () => {
+  const { addresses } = loadSeed(seedPath);
+  const provider = await makeProvider(ACCOUNTS);
+
+  john = await ContractAPI.userWrapper(ACCOUNTS[0], provider, addresses);
+  await john.approveCentralAuthority();
+}, 120000);
```
Comment (robert-e-davidson3) [inline packages/augur-test/src/tests/api/TransactionStatus.test.ts:20]: Can probably use a much smaller wait time like `15000`.

File: packages/augur-sdk/src/Augur.ts [hunk 3]
```diff
@@ -71,6 +71,8 @@ export class Augur<TProvider extends Provider = Provider> {
     this.trade = new Trade(this);
     this.market = new Market(this);
     this.events = new Events(this.provider, this.addresses.Augur);
+
+    this.registerTrandactionStatusEvents();
```
Comment (robert-e-davidson3) [inline packages/augur-sdk/src/Augur.ts]:
```suggestion
    this.registerTransactionStatusEvents();
```

File: packages/augur-sdk/src/Augur.ts [hunk 4]
```diff
@@ -208,4 +210,29 @@ export class Augur<TProvider extends Provider = Provider> {
   public async simulateTradeGasLimit(params: PlaceTradeDisplayParams): Promise<BigNumber> {
     return this.trade.simulateTradeGasLimit(params);
   }
+
+  private registerTrandactionStatusEvents() {
```
Comment (robert-e-davidson3) [inline packages/augur-sdk/src/Augur.ts]:
```suggestion
  private registerTransactionStatusEvents() {
```

File: packages/augur-sdk/src/Augur.ts [hunk 5]
```diff
@@ -151,9 +153,9 @@ export class Augur<TProvider extends Provider = Provider> {
     return Augur.connector.bindTo(f);
   }
 
-  public async on<T extends SubscriptionType>(eventName: SubscriptionEventName | string, callback: Callback): Promise<void> {
+  public async on(eventName: SubscriptionEventName | string, callback: Callback): Promise<void> {
     if (isSubscriptionEventName(eventName)) {
-      return Augur.connector.on(eventName, callback);
+      return Augur.connector.on(eventName, callback as Callback);
```
Comment (robert-e-davidson3) [inline packages/augur-sdk/src/Augur.ts:164]: Do you need to include `as Callback`? I thought specifying the type in the parameters would be enough.

**Review by robert-e-davidson3 — APPROVED** [review]
lgtm

**Review by bthaile — APPROVED** [review]
looks good, just wondering how to wire it up to the UI.

**Comment by brianosaurus** [comment]
damn @robert-e-davidson3 you've got a good eye. I'm impressed!

**Pull request merged**